src/apiClients/AirTableApiClient.ts

```diff
@@ -24,9 +35,61 @@ class AirTableApiClient {
       `interestCategories: ${interestCategories}`,
       `riverSections: ${riverSections}`,
     );
+    let organizations: Organization[] = [];
+
+    const promise: Promise<Organization[]> = new Promise((resolve, reject) => {
+      this.base('Organizations')
+        .select({
+          view: 'Grid view',
+        })
+        .eachPage(
+          (records: any[], fetchNextPage: any) => {
+            let filteredRecords: any[] = records;
```

Comment:
I think it'd be better to map the records into organizations immediately, so we don't have to deal with AirTable's `.get` function throughout. This will also help TypeScript do a better job here. Here's some code to get your started.

```ts

type TFieldName =
  | 'Name'
  | 'Description'
  | 'Activity'
  | 'Interest Categories'
  | 'River Section'
  | 'Phone Number'
  | 'URL'
  | 'Email';

interface Record {
  get: (fieldName: TFieldName) => string | string[];
}

function recordToOrganization(record: Record): Organization {
  return {
    name: record.get('Name') as string,
    description: record.get('Description') as string,
    activity: record.get('Activity') as string,
    interestCategories: record.get('Interest Categories') as string[],
    riverSection: record.get('River Section') as string,
    phoneNumber: record.get('Phone Number') as string,
    url: record.get('URL') as string,
    email: record.get('Email') as string,
  };
}
// your function in .eachPage should look something like this now
const func = (records: Record[], fetchNextPage: Function) => {
  let filteredOrganizations = records.map(recordToOrganization);
  // modify the rest
};

```

Reply:
First filter for undefined records, then map all records into Organization objects, then filter by interesteCategory/riverSection